services-directions-refresh/src/main/java/com/mapbox/api/directionsrefresh/v1/MapboxDirectionsRefresh.java

```diff
@@ -0,0 +1,150 @@
+package com.mapbox.api.directionsrefresh.v1;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.auto.value.AutoValue;
+import com.google.gson.GsonBuilder;
+import com.mapbox.api.directions.v5.DirectionsAdapterFactory;
+import com.mapbox.api.directionsrefresh.v1.models.DirectionsRefreshAdapterFactory;
+import com.mapbox.api.directionsrefresh.v1.models.DirectionsRefreshResponse;
+import com.mapbox.core.MapboxService;
+import com.mapbox.core.constants.Constants;
+import com.mapbox.core.utils.ApiCallHelper;
+
+import retrofit2.Call;
+
+/**
+ * The directions refresh API allows a route to be refreshed via it's annotations. The
+ * refreshEnabled parameter would have had to have been specified as true in the original
+ * directions request for a refresh to be possible.
+ *
+ * @since 4.4.0
+ */
+@AutoValue
+public abstract class MapboxDirectionsRefresh extends MapboxService<DirectionsRefreshResponse,
+  DirectionsRefreshService> {
+
+  protected MapboxDirectionsRefresh() {
+    super(DirectionsRefreshService.class);
+  }
+
+  @Override
+  protected Call<DirectionsRefreshResponse> initializeCall() {
+    return getService().getCall(
+      ApiCallHelper.getHeaderUserAgent(clientAppName()),
+      requestId(),
+      routeIndex(),
+      legIndex(),
+      accessToken()
+    );
+  }
+
+  abstract String requestId();
+
+  abstract String routeIndex();
+
+  abstract String legIndex();
+
+  abstract String accessToken();
+
+  @Nullable
+  abstract String clientAppName();
+
+  @Override
+  protected GsonBuilder getGsonBuilder() {
+    return super.getGsonBuilder()
+      .registerTypeAdapterFactory(DirectionsRefreshAdapterFactory.create())
+      .registerTypeAdapterFactory(DirectionsAdapterFactory.create());
+  }
+
+  /**
+   * Build a new {@link MapboxDirectionsRefresh} builder with default initial values
+   *
+   * @return a {@link Builder} for creating a default {@link MapboxDirectionsRefresh}
+   * @since 4.4.0
+   */
+  public static Builder builder() {
+    return new AutoValue_MapboxDirectionsRefresh.Builder()
+      .baseUrl(Constants.BASE_API_URL)
+      .routeIndex(0)
```

Comment:
I'm getting a compilation error when running this because route / leg index are defined as `String`s

Reply:
@danesfeder this is updated now. Also it's running on production now so the sample test should pass